File: Java/LLD/src/Problems/ATM/AccountFactory.java
package Problems.ATM;

public class AccountFactory {
    public static Account createAccount(String type, String accountNumber, double balance) {
        if ("savings".equalsIgnoreCase(type)) {
            return new SavingsAccount(accountNumber, balance);
        } else if ("current".equalsIgnoreCase(type)) {
            return new CurrentAccount(accountNumber, balance);
        } else {
            throw new IllegalArgumentException("Invalid account type: " + type);
        }
    }
}

File: Low Level Design (LLD)/Problems/ATM/Cpp/src/AccountFactory.cpp
#include "../include/AccountFactory.h"
#include "../include/SavingsAccount.h"
#include "../include/CurrentAccount.h"
#include <stdexcept>

Account* AccountFactory::createAccount(const std::string& type, const std::string& accNo, double balance) {
    if (type == "savings") return new SavingsAccount(accNo, balance);
    if (type == "current") return new CurrentAccount(accNo, balance);
    throw std::invalid_argument("Unknown account type");
}
